crates/ruff_linter/src/codes.rs

```diff
@@ -424,8 +427,8 @@ pub fn code_to_rule(linter: Linter, code: &str) -> Option<(RuleGroup, Rule)> {
         (Flake8Annotations, "001") => (RuleGroup::Stable, rules::flake8_annotations::rules::MissingTypeFunctionArgument),
         (Flake8Annotations, "002") => (RuleGroup::Stable, rules::flake8_annotations::rules::MissingTypeArgs),
         (Flake8Annotations, "003") => (RuleGroup::Stable, rules::flake8_annotations::rules::MissingTypeKwargs),
-        (Flake8Annotations, "101") => (RuleGroup::Stable, rules::flake8_annotations::rules::MissingTypeSelf),
-        (Flake8Annotations, "102") => (RuleGroup::Stable, rules::flake8_annotations::rules::MissingTypeCls),
+        (Flake8Annotations, "101") => (RuleGroup::Deprecated, rules::flake8_annotations::rules::MissingTypeSelf),
+        (Flake8Annotations, "102") => (RuleGroup::Deprecated, rules::flake8_annotations::rules::MissingTypeCls),
```

Comment:
I'm less certain on `ANN102`. I would never enable it myself, but could it have value?

Reply:
I can't see why. It think the annotations here are either `self: Self` or `cls: type[Self]`. Are there situations in which they would differ for the `cls` case?

Reply:
PEP-673 has a really annoying restriction that says you can't use `typing.Self` in metaclasses, so you have to use a custom type variable for `__new__` methods in metaclasses: https://github.com/python/typeshed/blob/3881b1b81a8118aa6fc2d2e9616c8f541e5101ed/stdlib/builtins.pyi#L194-L196

You also might need to use a custom type variable for `cls` if you can't use `typing.Self` for whatever reason -- maybe you support older Pythons and you can't declare a dependency on `typing_extensions`?

But I agree that this seems like a pretty pointless rule, honestly. The vast majority of the time, you can leave these unannotated, and no type checker is going to care if you leave them unannotated. If you need to annotate `cls`, you'll know that you need to do so without a lint rule telling you about it, I think